### DISCUSSION TITLE
Subscribe not emitting

### DISCUSSION THREAD

**Discussion by MaraSgit** [pocketbase/pocketbase#2025] [Q&A]
Hello, i use Pocketbase in my JS application and when i tried subscribe to collection nothing happens. There is no err in console (in my application)
Code:
```
      client.pb
        .collection("providedServers")
        .subscribe("*", (e) => {
          console.log(e.record);
        })
        .catch((err) => {
          console.log(err);
        });
``` 
When something happens just silence in console. Anybody have some fix?

**Comment by ganigeorgiev** [comment]
You'll receive events only if:

- there are record changes in the `providedServers` collection (create/update/delete)
- your client satisfies the List API rule of the `providedServers` collection

**Reply by MaraSgit** [reply]
> You'll receive events only if:
> 
> * there are record changes in the `providedServers` collection (create/update/delete)
> * your client satisfies the List API rule of the `providedServers` collection

My client is logged as admin and i'm creating and deleting records in the collection and still nothing

**Reply by ganigeorgiev** [reply]
@MaraSgit I've just tested it locally and I'm not able to reproduce it.

Could you export your collections structure ("Admin UI > Settings > Export Collections") and provide a more detailed code sample (or even better - a minimal reproducible repo)?

**Reply by ganigeorgiev** [reply]
The only things that I can think of that could prevent receiving events as admin:

- typo in the collection name
- have multiple open tabs connecting with http/1 (_see the [MDN EventSource warning](https://developer.mozilla.org/en-US/docs/Web/API/EventSource); the above limitation is not an issue when PocketBase is started with the `--https` flag since http/2 is served_)

**Reply by MaraSgit** [reply]
> The only things that I can think of that could prevent receiving events as admin:
> 
> * typo in the collection name
> * have multiple open tabs connecting with http/1 (_see the [MDN EventSource warning](https://developer.mozilla.org/en-US/docs/Web/API/EventSource); the above limitation is not an issue when PocketBase is started with the `--https` flag since http/2 is served_)

So when i'm connected to the admin ui i can't recieve events? Bcs there is no way of https since i running it on my vps which have just ip. Or am i wrong? And there is no typo in name, checked

**Reply by ganigeorgiev** [reply]
> So when i'm connected to the admin ui i can't recieve events?

No, when using http you can open from a single browser max ~6 persistent connections per domain. From the above linked MDN warning:

>  **This limit is per browser + domain, so that means that you can open 6 SSE connections across all of the tabs to www.example1.com and another 6 SSE connections to www.example2.com.**

If that's not the case, please provide reproducible steps or at least details about your setup. If you are using nginx as reverse proxy, please check the example config in https://pocketbase.io/docs/going-to-production/#using-reverse-proxy.

**Reply by MaraSgit** [reply]
So i have my pocketbase hosted on OCI vps, i have opened one browser window where i have the admin UI. I have nodejs v18 with pocketbase sdk installed and few other packages. Then i have this code in main file
```
const pb = new PocketBase("http://ip:8090/");
pb.admins.authWithPassword("email, "pass");
client.pb = pb;
```
(client is passed to other files and its instance of `oceanic.js`)
And then i have this code       
```
client.pb
        .collection("providedServers")
        .subscribe("*", (e) => {
          console.log(e.record);
        })
        .catch((err) => {
          console.log(err);
        });
```
that is in one of the files that run after full startup

**Reply by ganigeorgiev** [reply]
There are too many external factors and you are not providing enough details to help you debug it.

If the client is initialized in a node env, do you have `EventSource` polyfill as noted in https://github.com/pocketbase/js-sdk#nodejs-via-npm?

Are you able to reproduce the issue locally, aka. directly connecting to a local PocketBase instance in a browser?

**Reply by MaraSgit** [reply]
> There are too many external factors and you are not providing enough details to help you debug it.
> 
> If the client is initialized in a node env, do you have `EventSource` polyfill as noted in https://github.com/pocketbase/js-sdk#nodejs-via-npm?
> 
> Are you able to reproduce the issue locally, aka. directly connecting to a local PocketBase instance in a browser?

Yea the client is running in node env, yea i have EventSource installed but i never tried to replicate that, i will try. Give me a moment

**Reply by MaraSgit** [reply]
I tested it and on localhost it is working fine @ganigeorgiev

**Reply by ganigeorgiev** [reply]
I've also just tested it in a node 18 env (with PocketBase 0.13.2 and JS SDK 0.12.1) and it is working fine for me.

```js
import PocketBase from "pocketbase";
import eventsource from "eventsource";

global.EventSource = eventsource;

const pb = new PocketBase("http://127.0.0.1:8090");

await pb.admins.authWithPassword("test@example.com", "1234567890")

pb.collection("example").subscribe("*", (e) => {
    console.log(e);
})
```

There is something strange happening in your environment but I'm not sure how to reproduce it.
You can try starting PocketBase with the `--debug` flag and check whether there are "Realtime connection established/closed" logs. You can also try isolating one by one any external component in your system to narrow down the problematic part.
